### ISSUE TITLE
Exception running example from "Create a Windows Service using BackgroundService"

### ISSUE THREAD

**Issue by nbass-tech** [dotnet/docs#28503]
Edit: My post did not seem to get linked with the document I was referring to, here it is for reference:
Content: [Create a Windows Service using BackgroundService](https://docs.microsoft.com/en-us/dotnet/core/extensions/windows-service)

I've tried to create and build this project to the letter in VS2022 and also JetBrains Rider and I get the same exception whenever I try to run the executable or start it as a system service:

```
D:\Projects\WindowsService\WindowsService\bin\Release\net6.0\win-x64\publish>WindowsService.exe
Unhandled exception. System.InvalidOperationException: Unable to resolve service for type 'App.WindowsService.JokeService' while attempting to activate 'App.WindowsService.WindowsBackgroundService'.
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.CreateArgumentCallSites(Type implementationType, CallSiteChain callSiteChain, ParameterInfo[] parameters, Boolean throwIfCallSiteNotFound)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.CreateConstructorCallSite(ResultCache lifetime, Type serviceType, Type implementationType, CallSiteChain callSiteChain)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.TryCreateExact(ServiceDescriptor descriptor, Type serviceType, CallSiteChain callSiteChain, Int32 slot)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.TryCreateEnumerable(Type serviceType, CallSiteChain callSiteChain)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.CreateCallSite(Type serviceType, CallSiteChain callSiteChain)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.GetCallSite(Type serviceType, CallSiteChain callSiteChain)
   at Microsoft.Extensions.DependencyInjection.ServiceProvider.CreateServiceAccessor(Type serviceType)
   at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd(TKey key, Func`2 valueFactory)
   at Microsoft.Extensions.DependencyInjection.ServiceProvider.GetService(Type serviceType, ServiceProviderEngineScope serviceProviderEngineScope)
   at Microsoft.Extensions.DependencyInjection.ServiceProvider.GetService(Type serviceType)
   at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetService[T](IServiceProvider provider)
   at Microsoft.Extensions.Hosting.Internal.Host.StartAsync(CancellationToken cancellationToken)
   at Microsoft.Extensions.Hosting.HostingAbstractionsHostExtensions.RunAsync(IHost host, CancellationToken token)
   at Microsoft.Extensions.Hosting.HostingAbstractionsHostExtensions.RunAsync(IHost host, CancellationToken token)
   at Program.<Main>$(String[] args) in D:\Projects\WindowsService\WindowsService\Program.cs:line 14
   at Program.<Main>(String[] args)
```

I've gone over the article multiple times, but am I possibly missing something that might cause this?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: ed148088-4550-2788-0756-390eab15e3f6
* Version Independent ID: 64d5cdf8-e50f-0c89-7f16-82645ba6383e
* Content: [Create a Windows Service using BackgroundService - .NET](https://docs.microsoft.com/en-us/dotnet/core/extensions/windows-service)
* Content Source: [docs/core/extensions/windows-service.md](https://github.com/dotnet/docs/blob/main/docs/core/extensions/windows-service.md)
* Product: **dotnet-fundamentals**
* GitHub Login: @IEvangelist
* Microsoft Alias: **dapine**

**Comment by nbass-tech** [comment]
After scouring the web I came across this blog post:  Content: [Creating a Windows Service with C#/.NET5 - #ifdef Windows](https://devblogs.microsoft.com/ifdef-windows/creating-a-windows-service-with-c-net5/).  From that I got the idea of calling `AddSingleton` in the service configuration portion of the Program class and that seems to have resolved the issue.  Is this an omission/error in the docs?

```
using App.WindowsService;

using IHost host = Host.CreateDefaultBuilder(args)
    .UseWindowsService(options =>
    {
        options.ServiceName = ".NET Joke Service";
    })
    .ConfigureServices(services =>
    {
        services.AddHostedService<WindowsBackgroundService>();
        services.AddSingleton<JokeService>();
    })
    .Build();

await host.RunAsync();
```

**Comment by softwareliberationarmy** [comment]
@nbass-tech  it is. I ran into the same issue. The sample code for Progam.cs had omitted any kind of registration of the JokeService class.

**Comment by gewarren** [comment]
@IEvangelist Can you put this one on your radar?

**Comment by IEvangelist** [comment]
This section titled [Rewrite the `Program` class](https://docs.microsoft.com/dotnet/core/extensions/windows-service#rewrite-the-program-class) shows the `JokeService` being registered as a singleton.

![image](https://user-images.githubusercontent.com/7679720/186699232-1c2b11bd-7f7c-4368-a5e5-aae68c22663b.png)

This has been this way in the article since Mar 7th, 2022 and was addressed in #28522. This isn't an issue, unless I'm overlooking something. I'm going to close this for now, but please let me know if disagree @nbass-tech or @softwareliberationarmy
